Ginger/GingerCore/ALM/JIRA/Bll/JiraExportManager.cs

```diff
@@ -47,19 +47,131 @@ public JiraExportManager(JiraRepository.JiraRepository jiraRep)
+                if (exportedDefects.Count > 0)
+                {
+                    for (var a = 0; a < exportedDefects.Count; a++)
+                    {
+                        if (exportedDefects[a].DataResult != null)
+                        {
+                            AttachScreenshotsToDefects(screenshots[a], exportedDefects[a].DataResult.key);
+                            defectsOpeningResults.Add(defectsForOpening.ElementAt(a).Key, exportedDefects[a].DataResult.key);
+                        }
+                        else
+                        {
+                            Reporter.ToLog(eLogLevel.ERROR, "Failed to create to defect -" + exportedDefects[a].AuthenticationResponseObj.ErrorDesc);
+                        }
+                    }
+                }
             }
             return defectsOpeningResults;
         }
 
+        /// <summary>
+        /// Check for if defect is already exist with given summury value
+        /// if exist return id of the first defect
+        /// </summary>
+        /// <param name="summary">string summary value</param>
+        /// <returns></returns>
+        string CheckIfDefectExist(string summary)
+        {
+            //Search for exist based on summary
+            WhereDataList filterData = new WhereDataList();
+            filterData.Add(new WhereData() { Name = "summary", Values = new List<string>() { summary }, Operator = WhereOperator.And });
+
+            //and based on status, fetch statuses from JiraSettings.json 
+            DefectStatusColl jiraDefectStatuses = null;
+            JiraRepository.Helpers.JiraHelper jiraHelper = new JiraRepository.Helpers.JiraHelper();
+            if (jiraHelper.TryGetJiraDefectStatuses(out jiraDefectStatuses))
+            {
+                foreach (DefectStatus defectStatus in jiraDefectStatuses)
+                {
+                    ComparisionOperator comparisionOperator = GetComparisionOperator(defectStatus.Operator.ToLower());
+                    filterData.Add(new WhereData() { Name = "Status", Values = defectStatus.Status.ToList(), Operator = WhereOperator.And, ComparisionOperator = comparisionOperator });
+                }
+            }
+
+            var issues = jiraRepObj.GetJiraIssues(ALMCore.DefaultAlmConfig.ALMUserName, ALMCore.DefaultAlmConfig.ALMPassword, ALMCore.DefaultAlmConfig.ALMServerURL, ALMCore.DefaultAlmConfig.ALMProjectKey, ResourceType.DEFECT, filterData);
+            if (issues.DataResult.Count > 0)
+            {
+                return issues.DataResult.FirstOrDefault().key.ToString();
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        public ComparisionOperator GetComparisionOperator(string compareOperator)
+        {
+            ComparisionOperator comparisionOperator = ComparisionOperator.EqualTo;
+            switch (compareOperator)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6043077)